### PR TITLE
docs: update quickstart and add new change

### PR DESCRIPTION
**Pull request by JadenSimon**
Update the docs and add new change. ~~I put 'Go (1.14, 1.15, 1.16)' instead of 'Go (1.x)' since 'Go1.x' is pretty misleading.~~

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinmk3 — APPROVED** [review]

**Comment by bryceitoc9** [comment]
> I put 'Go (1.14, 1.15, 1.16)' instead of 'Go (1.x)' since 'Go1.x' is pretty misleading.

1.x is consistent with Lambda, SAM, and the other runtimes we've added. Since this all runs in a container anyway, I'm thinking we should stick with that (are those Go versions necessary for the Go extension? If so, mark that in the toolchain AWS docs)

**Comment by JadenSimon** [comment]
SAM CLI would be unable to build unoptimized Go binaries past Go1.13 due its reliance on Go modules. That would be the absolute minimum version currently, and that's only if the user has an older version of the Delve source code in their GOPATH. Saying we support 'Go1.x' makes it seem like we support all versions of Go, which isn't true at all.

File: README.quickstart.vscode.md [hunk 1]
```diff
@@ -94,6 +94,7 @@ contextual documentation, as shown below.
 The Toolkit _local SAM debugging_ feature supports these runtimes:
```
Comment (justinmk3) [inline README.quickstart.vscode.md:94]:
```suggestion
The Toolkit _local SAM debugging_ feature supports these Lambda runtimes:
```

**Comment by justinmk3** [comment]
mention https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/setup-toolchain.html

**Comment by JadenSimon** [comment]
Changed to 'Go 1.x'. The 'Configuring your toolchain' docs will make it very clear that Go 1.14 is required for debugging Go lambdas.

File: README.quickstart.vscode.md [hunk 2]
```diff
@@ -91,14 +91,15 @@ contextual documentation, as shown below.
...
+The Toolkit _local SAM debugging_ feature supports these Lambda runtimes:
 
 -   C# (.NET Core 2.1, 3.1; .NET 5.0)
+-   Go (1.x)
 -   Java (8, 8.al2, 11)
 -   JavaScript (Node.js 10.x, 12.x, 14.x)
 -   Python (2.7, 3.6, 3.7, 3.8)
 
-For more information see [Working with AWS Serverless Applications](https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/serverless-apps.html) in the user guide.
+For more information see [Working with AWS Serverless Applications](https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/serverless-apps.html) in the user guide. To get setup for local debugging, see [Configuring your toolchain](https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/setup-toolchain.html).
```
Comment (justinmk3) [inline README.quickstart.vscode.md:102]: it's also mentioned in `#setup` in this doc doc https://github.com/aws/aws-toolkit-vscode/blob/7f3b9298adc95f2c996fc3dc69856c7d3f4c2139/README.quickstart.vscode.md#setup

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**